src/main/java/com/shapesecurity/shift/es2017/utils/D2A.java

```diff
@@ -159,7 +159,7 @@ public static String shortD2a(double n) {
             }
             return "0"; // Not reached
         } else if (n % 1 == 0) {
-            if (n > 1e15 && n < 1e20) {
+            if (n > 1e15 && n < Math.pow(2, 63)) {
```

Comment:
It looks like the lower bound is looser than it needs to be, too.

```java
if (n >= 1e12 && n < 0x8000000000000000 /* 2^63 */) {
```

Reply:
This is java, we cannot represent an integer >= 2^63 `0x8000000000000000`, hence the usage of a double. I will change the lower bound.

Reply:
Ugh I assumed it would treat it as a double. Either way, works for me.